### DISCUSSION TITLE
Is there a way to toggle the scrollback with a single keyboard shortcut?

### DISCUSSION THREAD

**Discussion by carlsmedstad** [kovidgoyal/kitty#5576] [Q&A] [answered]
I'm looking into replicating the features I use in Alacritty and I've hit a dead end regarding the scrollback pager.

I've found the keyboard shorcut `ctrl+shift+h` which achieves part of what I want to do: Open the scrollback in my pager (Neovim). However, in order to close the pager and get back to the terminal I'm required to exit Neovim the regular way, with `:q`. Is there a way to map `ctrl+shift+h` to actually toggle the scrollback pager? I.e. if I'm in the terminal it will open the pager, If I'm in the pager it will close it. This will save a couple of keystrokes and is more ergonomic in my opinion.

Thanks!

**Comment by kovidgoyal** [comment]
map whatever key you want to quit in neovim, you can pass any command
line arguments you like to the pager.

**Reply by carlsmedstad** [reply]
I've not been able configure Neovim to exit on the same shortcut as the pager was opened with. When I press `ctrl+shift+h` in Neovim another instance of Neovim opens. It seems like Kitty acts on the shortcut before it gets passed to Neovim.

**Comment by kovidgoyal** [comment]
Dont use the same shortcut, use a different one. Like "q". Its much
easier to press than ctrl+shift+h.

If you really want to use the *same* shortcut write a kitten for it, the
kitten can examine the currently active window and if it s a pager
window, close it.

**Comment by page-down** [comment, marked as answer]
You can write a kitten to perform different actions depending on the conditions.
For example, the following kitten sends `ZQ` to exit when the scrollback pager (vim/neovim) is on.

~/.config/kitty/toggle_scrollback_pager.py
```python
from typing import List, Optional

from kitty.boss import Boss
from kittens.tui.handler import result_handler


def main(args: List[str]) -> None: 
    pass


@result_handler(no_ui=True)
def handle_result(args: List[str], data: Optional[str], target_window_id: int, boss: Boss) -> None:
    w = boss.window_id_map.get(target_window_id)
    if w is None:
        return
    if not w.screen.is_main_linebuf() and w.title == 'History':
        return w.send_text('all', b'ZQ')
    return w.show_scrollback()
```

```conf
map f1 kitten toggle_scrollback_pager.py
```

**Reply by carlsmedstad** [reply]
Fantastic, that was exactly what I was looking for. Much appreciated, thanks!
